src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -305,6 +305,7 @@ public Path exportProgrammingExerciseRepositories(ProgrammingExercise exercise,
      */
     public Optional<File> exportInstructorRepositoryForExercise(long exerciseId, RepositoryType repositoryType, List<String> exportErrors) {
         Path outputDir = fileService.getUniquePath(repoDownloadClonePath);
+        fileService.scheduleForDirectoryDeletion(outputDir, 5);
```

Comment:
Can you create a method combining these two and maybe even make getUniquePath private?